### ISSUE TITLE
stan_mvmer does not work with more than 3 outcomes

### ISSUE THREAD

**Issue by bgoodri** [stan-dev/rstanarm#248]
#### Summary:

@sambrilleman : There is an error from the R code when there is more than 3 outcomes

#### Description:

See
http://discourse.mc-stan.org/t/stan-mvmer-error-when-using-4-outcomes-ok-for-3-or-fewer/3078/3?u=bgoodri

#### Reproducible Steps:

```
fit = stan_mvmer(formula=list(mpg  ~ (1|cyl), disp ~ (1|cyl), hp ~ (1|cyl), qsec ~ (1|cyl)),
                 data=mtcars, seed=1, chains=1)
```
#### RStanARM Version:

2.17.2

#### R Version:

3.4.x

#### Operating System:

Debian

**Comment by sambrilleman** [comment]
Actually, yeah there is a hard limit on the number of outcomes.

stan_mvmer (and stan_jm) are limited to a maximum of 3 outcomes, and maximum of 2 grouping factors. It appears I included a stop() for the latter (see here) but not for the former.

Hard coding these massively sped up the code. Without it, the design matrices for the separate outcomes needed to be diagonally bound and then either split up later when constructing the linear predictor for the model (which may be reasonable), or the linear predictor needed to be constructed using the single block diagonal design matrix (which is slow). If the sparse matrix multiplication in Stan is sped up in the future then it might be reasonable to relax this restriction, so that the user can specify any number of outcomes & grouping factors, but for now I thought the speed advantages seemed too great (particularly for stan_jm, which can be slow in any case).

**I need to add a stop to either the validate_arg function, or somewhere else in stan_jm / stan_mvmer.**

**Comment by bgoodri** [comment]
OK

**Comment by sambrilleman** [comment]
@jgabry No unfortunately not. This bug fix and a couple of other small ones are on this branch:

https://github.com/stan-dev/rstanarm/tree/jm-updates-and-fixes

But that branch also now has some more extensive changes to the JM vignette that I haven't finished, and so it isn't ready to be pushed to the master. 

I'll work out how to tidy this up and send a pull request with just the small bug fixes asap, i.e. today, so that they can go out with the new release.

**Comment by sambrilleman** [comment]
@jgabry Sorry I've run out of time tonight. I'll pick it up first thing tomorrow morn.

When are you planning to push to CRAN?

**Comment by jgabry** [comment]
Nice thanks Sam!

**Comment by jgabry** [comment]
fixed in #269
